### DISCUSSION TITLE
Custom postgres port, e.g. 5435 in Docker?

### DISCUSSION THREAD

**Discussion by abdessalaam** [chatwoot/chatwoot#8561] [Self hosted]
My port 5432 is already used, so I have changed docker-compose: 
``` 
    ports:
      - '127.0.0.1:5435:5435'
``` 

and added a variable to .env `POSTGRES_PORT=5435` to use 5435 for Postgres instead.

However, I cannot connect to postgres. The error I get is: 
``` 
chatwoot-rails-1     | 2023-12-15T01:42:43.851511281Z + pg_isready -h postgres -p 5435 -U postgres
chatwoot-rails-1     | 2023-12-15T01:42:43.853456392Z postgres:5435 - no response
chatwoot-rails-1     | 2023-12-15T01:42:43.853710801Z + sleep 2
``` 

If I try: 
``` 
    ports:
      - '127.0.0.1:5435:5432'
``` 

Then the error is: 
``` 
chatwoot-rails-1     | 2023-12-15T01:56:37.844195863Z /gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_adapters/postgresql_adapter.rb:83:in `rescue in new_client': There is an issue connecting to your database with your username/password, username: postgres. (ActiveRecord::DatabaseConnectionError)
``` 

I would appreciate any help. Thanks!

**Comment by vishnu-narayanan** [comment]
Hey, the steps you have shared are correct.

1. Modify the postgres container host port on docker-compose.production.yaml (    ports:
      - '127.0.0.1:5435:5432'
)
2. Add `POSTGRES_PORT` env variable to the desired port on host. 

This is working correctly since the logs show rails trying to connect to 5435.

From the logs shared ,rails is not able to connect to postgres via port 5435. Could you try debugging that? SSH to the rails container and see if you can connect to postgres container.
```
docker exec -it $(basename $(pwd))-rails-1 sh
psql -h postgres -p 5435 -U postgres
```

**Reply by abdessalaam** [reply]
Thank you. It doesn't seem to connect via 5435, but it connects via 5432 although the authentication fails (password copied and pasted from my .env / docker-compose)

``` 
/app # psql -h postgres -p 5435 -U postgres
psql: error: connection to server at "postgres" (172.21.0.2), port 5435 failed: Connection refused
	Is the server running on that host and accepting TCP/IP connections?
/app # psql -h postgres -p 5432 -U postgres
Password for user postgres: 
psql: error: connection to server at "postgres" (172.21.0.2), port 5432 failed: FATAL:  password authentication failed for user "postgres"
```

**Comment by moreirarod** [comment]
Hi. I need help with that too! I can connect the database via another port but rails doesn't reach it.

**Comment by medoix** [comment]
Define a PGPORT environment variable within the .env file.

```
# Postgres Database config variables
# You can leave POSTGRES_DATABASE blank. The default name of
# the database in the production environment is chatwoot_production
# POSTGRES_DATABASE=
POSTGRES_HOST=postgres
POSTGRES_USERNAME=postgres
PGPORT=5435
POSTGRES_PASSWORD=
```
